include/seqan3/utility/views/type_reduce.hpp

```diff
@@ -0,0 +1,174 @@
+ * |                                  |                                   |                                        |
+ * | std::ranges::viewable_range      | *required*                        | *guaranteed*                           |
+ * | std::ranges::view                |                                   | *guaranteed*                           |
+ * | std::ranges::sized_range         |                                   | *preserved*                            |
+ * | std::ranges::common_range        |                                   | *preserved*                            |
+ * | std::ranges::output_range        |                                   | *preserved*                            |
+ * | seqan3::const_iterable_range     |                                   | *preserved*                            |
+ * |                                  |                                   |                                        |
+ * | std::ranges::range_reference_t   |                                   | std::ranges::range_reference_t<urng_t> |
+ *
+ * See the \link views views submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * ### Return type
+ *
+ * | `urng_t` (underlying range type)                                                                  | `rrng_t` (returned range type)  |
+ * |:-------------------------------------------------------------------------------------------------:|:-------------------------------:|
+ * | `std::ranges::view`                                                                               | `urng_t`                        |
+ * | `std::basic_string const &` *or* `std::basic_string_view`                                         | `std::basic_string_view`        |
+ * | `std::ranges::borrowed_range && std::ranges::sized_range && std::ranges::contiguous_range`        | `std::span`                     |
+ * | `std::ranges::borrowed_range && std::ranges::sized_range && std::ranges::random_access_range`     | `std::ranges::subrange`         |
+ * | *else*                                                                                            | `std::ranges::ref_view<urng_t>` |
+ *
+ * This adaptor is different from std::views::all in that it performs type erasure for some underlying ranges;
+ * std::views::all always returns `std::ranges::ref_view<urng_t>` or the type itself if it already is a view.
+ *
+ * ### Example
+ *
+* \include test/snippet/utility/views/type_reduce.cpp
+ *
+ * \hideinitializer
+ * \experimentalapi{Experimental since version 3.1.}
+ */
+inline constexpr auto type_reduce = detail::type_reduce_fn{};
+
+//!\}
+
+} // namespace seqan3::views
+
+namespace seqan3
+{
+/*!\brief Deduces the return value of seqan3::views::type_reduce.
+ * \details
+ * \noapi
+ * \sa https://en.cppreference.com/w/cpp/ranges/all_view
+ */
+template <typename t>
+using type_reduce_t = decltype(views::type_reduce(std::declval<t>()));
+}
```

Comment:
Can you re-add + deprecate `type_reduce_view`

Also, can you add a changelog entry for the rename? 
> Renamed `seqan3::type_reduce_view` to `seqan3::type_reduce_t`

(we will do the changelog entry for the header moves at the very end, but we should document the rename as separate entry)